Packs/Thycotic/Integrations/Thycotic/Thycotic.py

```diff
@@ -209,29 +209,18 @@ def userUpdate(self, id: str, **kwargs) -> str:
     def userDelete(self, id: str) -> str:
         return self._http_request("DELETE", url_suffix="/api/v1/users/" + str(id))
 
-    def getCredentials(self) -> list:
-        credentials = []
+    def getCredentials(self) -> str:
+        credentials = {}
+        secretID = self.searchSecretIdByName(self._credential_objects)
 
-        if self._is_fetch_credential and (len(self._credential_objects) != 0):
-            listArgs = (str(self._credential_objects)).split(",")
-            for key in listArgs:
-                object = {'name': key}
-                secretID = self.searchSecretIdByName(key)[0]
-                object['user'] = self.getUsernameById(secretID)
-                object['password'] = self.getPasswordById(secretID)
-                credentials.append(object)
+        if len(secretID) == 1:
+            credentials['name'] = self._credential_objects
+            credentials['user'] = self.getUsernameById(secretID[0])
+            credentials['password'] = self.getPasswordById(secretID[0])
+        else:
+            return ""
```

Comment:
I'm not sure I understand what you're going for here.
It seems to me like when there's more than one credential to return, nothing will be returned, and it only works when there is one credential to return.
Am I missing anything?

Reply:
@guykeller 
Yes you are right.
I am only returning one credential.
Server secret does not have a unique parameter for the secret other than the ID. A situation may arise when several secrets can have the same secret name and username values.
And when filling in the data for the fetch_credential command, an error situation will occur, which was shown in the last demo.

Reply:
Please see my other comment regarding the args